examples/step-67/step-67.cc

```diff
@@ -1465,6 +1501,288 @@ namespace Euler_DG
+
+                          auto rho_u_dot_n = w_m[1] * normal[0];
+                          for (unsigned int d = 1; d < dim; ++d)
+                            rho_u_dot_n += w_m[1 + d] * normal[d];
+
+                          bool at_outflow = false;
+
+                          Tensor<1, dim + 2, VectorizedArray<Number>> w_p;
+
+                          if (wall_boundaries.find(boundary_id) !=
+                              wall_boundaries.end())
+                            {
+                              w_p[0] = w_m[0];
+                              for (unsigned int d = 0; d < dim; ++d)
+                                w_p[d + 1] =
+                                  w_m[d + 1] - 2. * rho_u_dot_n * normal[d];
+                              w_p[dim + 1] = w_m[dim + 1];
+                            }
+                          else if (inflow_boundaries.find(boundary_id) !=
+                                   inflow_boundaries.end())
+                            w_p = evaluate_function(
+                              *inflow_boundaries.find(boundary_id)->second,
+                              phi_m.quadrature_point(q));
+                          else if (subsonic_outflow_boundaries.find(
+                                     boundary_id) !=
+                                   subsonic_outflow_boundaries.end())
+                            {
+                              w_p = w_m;
+                              w_p[dim + 1] =
+                                evaluate_function(*subsonic_outflow_boundaries
+                                                     .find(boundary_id)
+                                                     ->second,
+                                                  phi_m.quadrature_point(q),
+                                                  dim + 1);
+                              at_outflow = true;
+                            }
+                          else
+                            AssertThrow(false,
+                                        ExcMessage(
+                                          "Unknown boundary id, did "
+                                          "you set a boundary condition for "
+                                          "this part of the domain boundary?"));
+
+                          auto flux =
+                            euler_numerical_flux<dim>(w_m, w_p, normal);
+
+                          if (at_outflow)
+                            for (unsigned int v = 0;
+                                 v < VectorizedArray<Number>::size();
+                                 ++v)
+                              {
+                                if (rho_u_dot_n[v] < -1e-12)
+                                  for (unsigned int d = 0; d < dim; ++d)
+                                    flux[d + 1][v] = 0.;
+                              }
+
+                          phi_m.submit_value(-flux, q);
+                        }
```

Comment:
For all the code above, we should have an `always_inline` function that does the operation at a quadrature point. This should be exactly the same as in the respective face function.